2/contrib/jenkins/kube-slave-common.sh

```diff
@@ -149,8 +141,8 @@ function generate_kubernetes_config() {
       <addMasterProxyEnvVars>true</addMasterProxyEnvVars>
       <serverCertificate>${crt_contents}</serverCertificate>
       <namespace>${PROJECT_NAME}</namespace>
-      <jenkinsUrl>http://${JENKINS_SERVICE_HOST}:${JENKINS_SERVICE_PORT}</jenkinsUrl>
-      <jenkinsTunnel>${JNLP_HOST}:${JNLP_PORT}</jenkinsTunnel>
+      <jenkinsUrl>http://${JENKINS_SERVICE_NAME}:${JENKINS_SERVICE_PORT}</jenkinsUrl>
+      <jenkinsTunnel>${JNLP_SERVICE_NAME}:${JNLP_PORT}</jenkinsTunnel>
```

Comment:
my main concern with this PR is here:  you're going to set the jenkinsurl to "jenkins:PORT".  Since you didn't put the namespace in the service name, no one outside this namespace will be able to resolve that url.

When we used the service ip, it would work.

you're going to need to include the namespace in the service name variables you are generating so they can be resolved outside the local namespace.

Reply:
(it's possible that because this config is only used for the k8s plugin, it doesn't have to resolve outside the local project, i'd like @gabemontero to weigh in on that)

Reply:
Yes, I get your point. If the namespace of the agents differs in the <clouds> configuration then we need to use a qualified name like <service>.<ns>.svc.cluster.local where ns is the namespace of the Jenkins master.

I'll modify and push when I tested.

Reply:
Service name is actually resolved without need for further modifications.
This is the log from the agent when \<clouds\> configuration specifies another namespace.
The k8s plugin resolves the service names and supply the full DNS name to the agent at startup:

OPENSHIFT_JENKINS_JVM_ARCH='', CONTAINER_MEMORY_IN_MB='1536', using /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.i386/jre/bin/java and /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.i386/bin/javac
Downloading http://jenkins.demo-j.svc.cluster.local:80//jnlpJars/remoting.jar  ...

The Jenkins configuration should thereby only contain the short service name.

Reply:
Yes, the download line contain the IP address without my change. For example:

OPENSHIFT_JENKINS_JVM_ARCH='', CONTAINER_MEMORY_IN_MB='8796093022207', using /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.x86_64/jre/bin/java and /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.171-8.b10.el7_5.x86_64/bin/javac
--
  | Downloading http://172.30.193.155:80//jnlpJars/remoting.jar  ...